File: FizzBuzzKata/JavaSolution/FizzBuzzKata/FizzBuzz.java
package FizzBuzzKata;

public class FizzBuzz {
	private int boundingValue;

	public static void main(String[] args) {
		FizzBuzz fb = new FizzBuzz(50);
		FizzBuzz fb2 = new FizzBuzz(-50);
		fb.printResults();
		System.out.println("negative case:");
		fb2.printResults();
	}

	public FizzBuzz() {
	}

	public FizzBuzz(int boundingValue) {
		this.boundingValue = boundingValue;
	}

	public void printResults() {
		if (boundingValue > 0) {
			for (int i = 1; i <= boundingValue; i++) {
				System.out.println(evaluate(i));
			}
		} else {
			for (int i = -1; i >= boundingValue; i--) {
				System.out.println(evaluate(i));
			}
		}
	}

	public String evaluate(int n) {
		if (isMultipleOf3(n) && isMultipleOf5(n)) {
			return "FizzBuzz";
		} else if (isMultipleOf3(n)) {
			return "Fizz";
		} else if (isMultipleOf5(n)) {
			return "Buzz";
		} else {
			return Integer.toString(n);
		}
	}

	private boolean isMultipleOf5(int n) {
		return n % 5 == 0;
	}

	private boolean isMultipleOf3(int n) {
		return n % 3 == 0;
	}
}

File: FizzBuzzKata/CSharpSolution/FizzBuzz.cs
using System;

namespace FizzBuzzKata {
	public class FizzBuzz {
		private int boundingValue;

		static void Main() {
			FizzBuzz fb = new FizzBuzz(100);
			FizzBuzz fb2 = new FizzBuzz(-100);
			fb.printResults();
			Console.WriteLine("Negative case:");
			fb2.printResults();
		}

		public FizzBuzz() { }

		public FizzBuzz(int value) {
			this.boundingValue = value;
		}

		public void printResults() {
			if (boundingValue > 0) {
				for (int i = 1; i <= boundingValue; i++) {
					Console.WriteLine(eval(i));
				}
			} else {
				for (int i = -1; i >= boundingValue; i--) {
					Console.WriteLine(eval(i));
				}
			}
		}

		public string eval(int n) {
			string result = "";

			if (isMultipleOf3(n)) {
				result += "Fizz";
			}
			if (isMultipleOf5(n)) {
				result += "Buzz";
			}
			if (result == "") {
				result += n;
			}
			return result;
		}

		private static bool isMultipleOf3(int n) {
			return n % 3 == 0;
		}

		private static bool isMultipleOf5(int n) {
			return n % 5 == 0;
		}
	}
}
